kstreams/middleware/middleware.py

```diff
@@ -95,25 +91,40 @@ async def __call__(self, cr: types.ConsumerRecord) -> typing.Any:
             await self.cleanup_policy(exc)
 
     async def cleanup_policy(self, exc: Exception) -> None:
-        # always release the asyncio.Lock `is_processing` to
-        # stop or restart properly the `stream`
+        """
+        Execute clenup policicy according to to Stream configuration.
+
+        At this point we are inside the asyncio.Lock `is_processing`
+        as an event is being processed and an exeption has occured.
+        The Lock must be released to stop the Stream
+        (which must happen for any policy), then before re-raising
+        the exception the Lock must be acquire again to continue the processing.
+
+        Args:
+            exc (Exception): Any Exception that causes the Stream to crash
+
+        Raises:
+            exc: Exception
+        """
         self.stream.is_processing.release()
 
         if self.error_policy == StreamErrorPolicy.RESTART:
             await self.stream.stop()
-            logger.info(f"Restarting stream {self.stream}")
             await self.stream.start()
         elif self.error_policy == StreamErrorPolicy.STOP:
             await self.stream.stop()
+            # acquire `is_processing` Lock again to resume processing
+            # and avoid `RuntimeError: Lock is not acquired.`
+            await self.stream.is_processing.acquire()
```

Comment:
should the acquire happen before or after the stop? what would be the logic on each case?

Reply:
This PR should never happend. It was an error from my side using the githug UI. I solved this issue months ago. https://github.com/kpn/kstreams/commit/e5c65076f17bf6956a26cb67ef689c3f31fb5ebc